cpmpy/transformations/reification.py

```diff
@@ -88,15 +87,14 @@ def only_implies(constraints):
                 newcons.append(cpm_expr)
             else:
                 # BVar1 == BE0 :: ~BVar1 -> ~BE0, BVar1 -> BE0
-                newexprs = ((~a0).implies(recurse_negation(a1)), a0.implies(a1))
-                if isinstance(a1, GlobalConstraint):
-                    newcons.extend(newexprs)
```

Comment:
is it not worth skipping the retransform in the case of bv -> global ?

Reply:
it felt dangerous that if the 'original' a1 is global, that the `recurse_negation(a1)` in newexprs would not need further transformations?
it sounds like there must be cases where this is would happen...

or is there some implicit assumption that decompose_globals would have decomposed it? but still... an implication on a negated global?

Reply:
push down negation does not decompose, it just returns ~global(args) (after pushing down negation in the args)
So even calling push down negation is not required in the case of a global.

I guess in practise we won't encounter a global here because indeed the global constraint would have to be supported in reified context to make it this far without being decomposed.
Only choco supports reified globals (at the moment), and it does not us only_implies